### ISSUE TITLE
Containers with same boot.autostart.priority not started together

### ISSUE THREAD

**Issue by wociscz** [lxc/lxd#4213]
* Distribution: Ubuntu
 * Distribution version: 16.04.3 LTS
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.13.0-32-generic
   * LXC version: 2.0.8-0ubuntu1~16.04.2
   * LXD version: 2.21-0ubuntu2~16.04.1
   * Storage backend in use: dir

# Issue description

When i set same boot.autostart.priority to multiple containers, they are started one by one (by their name order).

# Steps to reproduce

 1. lxc config set firstcontainer boot.autostart.priority 100
 2. lxc config set secondbatch1 boot.autostart.priority 50
 3. lxc config set secondbatch2 boot.autostart.priority 50

then reboot lxd host, after reboot - firstcontainer start as first (it is ok), then secondbatch1 and after them secondbatch2. When secondbatch1 starting (state RUNNING), secondbatch2 is at state STOPPED. I think that secondbatch1 and secondbatch2 should start together.
If you have lxc config set default boot.autostart.delay 10, it is more "visible".

Maybe it is not a bug, but feature?
Regards,
Jan

**Comment by stgraber** [comment]
Yeah, I think it was done intentionally so that someone who doesn't use the priority feature can set a delay between containers start as otherwise the delay property would be effectively meaningless unless you also use priority.
